wandb/sync/sync.py

```diff
@@ -214,41 +245,34 @@ def run(self):
                     continue
                 if len(filtered_files) > 0:
                     sync_item = os.path.join(sync_item, filtered_files[0])
-            root_dir = os.path.dirname(sync_item)
-            # If we're syncing tensorboard, let's use a tmpdir
-            if tb_event_files > 0 and not sync_item.endswith(WANDB_SUFFIX):
-                root_dir = TMPDIR.name
+            sync_tb = self._setup_tensorboard(
+                tb_root, tb_logdirs, tb_event_files, sync_item
+            )
+            # If we're syncing tensorboard, let's use a tmp dir for images etc.
+            root_dir = TMPDIR.name if sync_tb else os.path.dirname(sync_item)
             sm = sender.SendManager.setup(root_dir)
+            if sync_tb:
+                self._send_tensorboard(tb_root, tb_logdirs, sm)
+                continue
 
-            if tb_root is not None:
-                if tb_event_files > 0 and sync_item.endswith(WANDB_SUFFIX):
-                    wandb.termwarn(
-                        "Found .wandb file, not streaming tensorboard metrics."
-                    )
-                else:
-                    print(
-                        "Found {} tfevent files in {}".format(tb_event_files, tb_root)
-                    )
-                    if len(tb_logdirs) > 3:
-                        wandb.termwarn(
-                            "Found {} directories containing tfevent files. "
-                            "If these represent multiple experiments, sync them "
-                            "individually or pass a list of paths."
-                        )
-                    self._send_tensorboard(tb_root, tb_logdirs, sm)
-                    continue
```

Comment:
The linter was complaining about the complexity of this method so I put this into a helper method.